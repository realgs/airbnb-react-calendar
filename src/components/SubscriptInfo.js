import React from 'react';

const SubscriptInfo = ({ lastModified, lastUpdate, clearDates }) => {
  return (
    <div className="subscript">
      <span className="subscript__info">
        Minimum stay varies
      </span>
      { lastUpdate && (
        <span className="subscript__update">
          Updated {lastUpdate} days ago
        </span>
      )
      }
      { lastModified && (
        <div className="subscript__clearDates">
          <button className="subscript__clearDates__button" onClick={clearDates}>Clear dates</button>
        </div>
        )
      }
    </div>
  );
};


export default SubscriptInfo;