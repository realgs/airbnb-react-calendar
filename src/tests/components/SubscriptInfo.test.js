import React from 'react';
import { shallow } from 'enzyme';
import SubscriptInfo from '../../components/SubscriptInfo';

test('Should render info & last update in SubscriptInfo correctly', () => {
  const wrapper = shallow(<SubscriptInfo lastModified={"30-01-2019"} lastUpdate={2} clearDates={SubscriptInfo.clearDates} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.subscript').exists()).toBeTruthy();
  expect(wrapper.find('.subscript__info').exists()).toBeTruthy();
  expect(wrapper.find('.subscript__update').exists()).toBeTruthy();
  expect(wrapper.find('.subscript__clearDates').exists()).toBeTruthy();
});

test('Should render SubscriptInfo with no data provided', () => {
  const wrapper = shallow(<SubscriptInfo lastModified={null} lastUpdate={null} clearDates={null} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('.subscript').exists()).toBeTruthy();
  expect(wrapper.find('.subscript__info').exists()).toBeTruthy();
  expect(wrapper.find('.subscript__clearDates').exists()).toBeFalsy();
  expect(wrapper.find('.subscript__update').exists()).toBeFalsy();
});