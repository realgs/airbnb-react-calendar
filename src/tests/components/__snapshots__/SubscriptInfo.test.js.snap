// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render SubscriptInfo with no data provided 1`] = `
<div
  className="subscript"
>
  <span
    className="subscript__info"
  >
    Minimum stay varies
  </span>
</div>
`;

exports[`Should render info & last update in SubscriptInfo correctly 1`] = `
<div
  className="subscript"
>
  <span
    className="subscript__info"
  >
    Minimum stay varies
  </span>
  <span
    className="subscript__update"
  >
    Updated 
    2
     days ago
  </span>
  <div
    className="subscript__clearDates"
  >
    <button
      className="subscript__clearDates__button"
    >
      Clear dates
    </button>
  </div>
</div>
`;
